app/js/app-utilities.js

```diff
@@ -2058,4 +2058,70 @@ appUtilities.launchWithModelFile = function() {
   }  
 }
 
+appUtilities.animateToOtherEnd = function(edge, mouse_position) {
+  
+  if(!edge.isEdge()){
+    return;
+  }
+
+  var cy = appUtilities.getActiveCy();
+  var bend_instance = cy.edgeBendEditing('get'); 
+
+  var source_node = edge.source();
+  var target_node = edge.target();
+  
+  var source_position = source_node.position();
```

Comment:
Instead of `source_position` and `target_position`, use `edge.sourceEndpoint` and `edge.targetEndpoint` Using position of nodes will result in wrong calculation of closest node when the size of node is big.

Reply:
I tried to use `edge.sourceEndpoint()` and `edge.targetEndpoint()` but it says those are not functions. `edge.sourceEndpoint` and `edge.targetEndpoint` return `undefined` as well.

Reply:
@ahmetcandiroglu it looks like `sourceEndpoint`and `targetEndpoint`functions are only available in Cytoscape.js v3.2.0 and up (we are using v3.1.1). Let's keep the current implementation for now.